core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -144,56 +158,40 @@ public Object visitGetCacheEntryCommand(InvocationContext ctx, GetCacheEntryComm
       }
    }
 
-   private Object visitGetCommand(InvocationContext ctx, AbstractDataCommand command,
-      boolean isGetCacheEntry) throws Throwable {
-      Object returnValue = invokeNextInterceptor(ctx, command);
-
-      //if the cache entry has the value lock flag set, skip the remote get.
-      CacheEntry entry = ctx.lookupEntry(command.getKey());
-      boolean skipRemoteGet = entry != null && entry.skipLookup();
-
-      // need to check in the context as well since a null retval is not necessarily an indication of the entry not being
-      // available.  It could just have been removed in the same tx beforehand.  Also don't bother with a remote get if
-      // the entry is mapped to the local node.
-      if (!skipRemoteGet && returnValue == null && ctx.isOriginLocal()) {
-         Object key = filterDeltaCompositeKey(command.getKey());
-         if (needsRemoteGet(ctx, command)) {
-            InternalCacheEntry ice = remoteGet(ctx, key, false, command);
-            if (ice != null) {
-               returnValue = ice.getValue();
-            }
-         }
-         if (returnValue == null && !ctx.isEntryRemovedInContext(command.getKey())) {
-            returnValue = localGet(ctx, key, false, command, isGetCacheEntry);
-         }
-      }
-      return returnValue;
-   }
-
-   protected void lockAndWrap(InvocationContext ctx, Object key, InternalCacheEntry ice, FlagAffectedCommand command) throws InterruptedException {
-      boolean skipLocking = hasSkipLocking(command);
-      long lockTimeout = getLockAcquisitionTimeout(command, skipLocking);
-      lockManager.acquireLock(ctx, key, lockTimeout, skipLocking);
-      entryFactory.wrapEntryForPut(ctx, key, ice, false, command, false);
-   }
-
    @Override
    public Object visitLockControlCommand(TxInvocationContext ctx, LockControlCommand command) throws Throwable {
       if (ctx.isOriginLocal()) {
          //In Pessimistic mode, the delta composite keys were sent to the wrong owner and never locked.
          final Collection<Address> affectedNodes = cdl.getOwners(filterDeltaCompositeKeys(command.getKeys()));
          ((LocalTxInvocationContext) ctx).remoteLocksAcquired(affectedNodes == null ? dm.getConsistentHash().getMembers() : affectedNodes);
-         log.tracef("Registered remote locks acquired %s", affectedNodes);
-         rpcManager.invokeRemotely(affectedNodes, command, rpcManager.getDefaultRpcOptions(true, DeliverOrder.NONE));
+         if (trace) {
+            log.tracef("Registered remote locks acquired %s", affectedNodes);
+         }
+         try {
+            rpcManager.invokeRemotely(affectedNodes, command, rpcManager.getDefaultRpcOptions(true, DeliverOrder.NONE));
+         } catch (SuspectException e) {
+            partitionHandlingManager.addPartialRollbackTransaction(command.getGlobalTransaction(), ((LocalTxInvocationContext) ctx).getRemoteLocksAcquired(), ctx.getLockedKeys(), e);
```

Comment:
This makes me realize I need to extend my ISPN-5274 fix to retry LockControlCommands as well :)